### DISCUSSION TITLE
Host discovery seems to break with multiple ports (-ps)

### DISCUSSION THREAD

**Discussion by yaabdala** [projectdiscovery/naabu#949] [General]
### Current Behavior:
When I list multiple ports with the `-ps` flag like so `-ps 80,443` Host discovery fails. Using only one flag, it works.

### Expected Behavior:
Host discovery will check all listed ports and if any are found listening, return host found

### Steps To Reproduce:
Using standard config, the following fails
`sudo /home/ouser/go/bin/naabu -host 203.83.216.0/23  -sn -ps 80,443 `

[Screenshot from 2023-06-12 14-29-15](https://github.com/projectdiscovery/naabu/assets/100624692/eabe535d-7a0f-499a-abe4-326741c570d1)

**Comment by Mzack9999** [comment]
I guess it can depend on network congestion. For example by increasing the number of retries (nmap has up to 10) the results tend to be stable:

```
$ sudo go run . -host 192.168.5.0/23  -sn -ps 80,443 -silent -retries 5 | wc
      12      12     156
$ sudo go run . -host 192.168.5.0/23  -sn -ps 80,443 -silent -retries 5 | wc
      12      12     156
$ sudo go run . -host 192.168.5.0/23  -sn -ps 80,443 -silent -retries 5 | wc
      12      12     156
```

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
